### PR TITLE
feat(bin): link bins and run scripts

### PR DESCRIPTION
**Pull request by zkat**
Fixes: #4

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -111,10 +115,18 @@ class Installer {
       )
       child.pending = BB.resolve()
       .then(() => extract.child(child.name, child, childPath, this.config))
-      .then(() => readJson(childPath, 'package.json'))
+      .then(() => readPkgJson(path.join(childPath, 'package.json')))
       .then(pkg => {
         return this.runScript('preinstall', pkg, childPath)
         .then(() => this.extractTree(child))
+        .then(() => binLink(pkg, childPath, false, {
+          ignoreScripts: this.config.lifecycleOpts.ignoreScripts,
```
Comment (mikesherov) [inline index.js]: There’s a few more options needed here that get passed from bin-links to gentleFS

File: index.js [hunk 2]
```diff
@@ -111,10 +115,18 @@ class Installer {
       )
       child.pending = BB.resolve()
       .then(() => extract.child(child.name, child, childPath, this.config))
-      .then(() => readJson(childPath, 'package.json'))
```
Comment (mikesherov) [inline index.js]: Can you delete the readJson function now?

File: index.js (hunk 1 above)
Comment (mikesherov) [reply index.js]: I have them in my local copy at work. Will send Monday, or you can figure it out :-). At least one is: `name: cipm`

**Review by mikesherov — CHANGES_REQUESTED** [review]
Looks good, just super minor stuff missing

File: index.js (hunk 2 above)
Comment (mikesherov) [reply index.js]: Nvm, you need it to read the lock and shrink wrap files!

**Comment by zkat** [comment]
Should be good to re-review. I don't think there's any missing opts here.

**Review by mikesherov — APPROVED** [review]

File: index.js [hunk 3]
```diff
@@ -111,10 +115,20 @@ class Installer {
...
       .then(pkg => {
         return this.runScript('preinstall', pkg, childPath)
         .then(() => this.extractTree(child))
+        .then(() => binLink(pkg, childPath, false, {
+          force: this.config.config.force,
+          ignoreScripts: this.config.lifecycleOpts.ignoreScripts,
+          log: this.log,
+          name: pkg.name,
+          pkgId: pkg.name + '@' + pkg.version,
+          prefix: this.prefix,
```
Comment (mikesherov) [inline index.js:128]: I'm not sure this is right, btw. opts.prefix is used for `man` linking, which is usually like `/usr/local` in npm I think. I should rename this property `manPrefix`. Probably ok to land for now, but we should fix this before launch.

```
function linkMans (pkg, folder, parent, gtop, opts, cb) {
  if (!pkg.man || !gtop || process.platform === 'win32') return cb()

  var manRoot = path.resolve(opts.prefix, 'share', 'man')
  opts.log.verbose('linkMans', 'man files are', pkg.man, 'in', manRoot)
```

**Pull request merged**